lib/Backends/BackendUtils.cpp

```diff
@@ -55,3 +55,105 @@ size_t glow::getValueOffset(
+      symbol.size = numBytes;
+      symbol.offset = addr;
+      symbol.type = *w->getType();
+      symbolTable.emplace(std::string(v->getName()), symbol);
+    }
+    constantMaxSize = constantAllocator->getMaxMemoryUsage();
+  }
+
+  // Compute the offsets for Placeholders.
+  if (placeholderAllocator) {
+    for (auto &v : F->getGraph()->getParent()->getPlaceholders()) {
+      // Get the WeightVar for each Placeholder to calculate offsets.
+      assert(isa<WeightVar>(F->getWeightForNode(v)) && "Expected WeightVar");
+      auto *w = cast<WeightVar>(F->getWeightForNode(v));
+      auto numBytes = w->getSizeInBytes();
+      size_t addr = placeholderAllocator->allocate(numBytes, w);
+      runtime::RuntimeSymbolInfo symbol;
+      symbol.offset = addr;
+      symbol.size = numBytes;
+      symbol.type = *w->getType();
+      symbolTable.emplace(std::string(v->getName()), symbol);
+    }
+    placeholderMaxSize = placeholderAllocator->getMaxMemoryUsage();
+  }
+  // Compute the offsets for Activations.
+  if (activationsAllocator) {
+    for (const auto &I : F->getInstrs()) {
+      if (auto *A = llvm::dyn_cast<AllocActivationInst>(&I)) {
+        auto numBytes = I.getSizeInBytes();
+        size_t addr = activationsAllocator->allocate(numBytes, A);
+        assert(!symbolTable.count(std::string(A->getName())) &&
+               "Allocation already made!");
+        runtime::RuntimeSymbolInfo symbol;
+        symbol.offset = addr;
+        symbol.size = numBytes;
+        symbol.type = *A->getType();
+        symbolTable.emplace(std::string(A->getName()), symbol);
+        continue;
+      }
+
+      if (auto *TV = llvm::dyn_cast<TensorViewInst>(&I)) {
+        // Calculate and store the length of the offset into the base, using the
+        // source of the tensorview.
+        assert(!symbolTable.count(std::string(TV->getName())) &&
+               "Allocation already made!");
+        size_t offsetLength =
+            TV->getOffsets().empty() ? 0 : TV->getOffsets()[0];
+        auto *tvSource = TV->getSrc();
+        if (tvSource->dims().size() > 1) {
+          for (size_t i = 1; i < tvSource->dims().size(); ++i) {
+            offsetLength *= tvSource->dims()[i];
+          }
+        }
+        assert(symbolTable.count(std::string(tvSource->getName())) &&
+               "Source allocation not found!");
+        runtime::RuntimeSymbolInfo symbol;
+        symbol.offset = getValueOffset(tvSource, symbolTable) +
+                        (offsetLength * TV->getType()->getElementSize());
+        symbol.size = TV->getSizeInBytes();
+        symbol.type = *TV->getType();
+        symbolTable.emplace(std::string(TV->getName()), symbol);
+        continue;
+      }
+
+      if (auto *D = llvm::dyn_cast<DeallocActivationInst>(&I)) {
+        auto *A = D->getAlloc();
+        assert(symbolTable.count(std::string(A->getName())) &&
+               "Invalid deallocation!");
+        activationsAllocator->deallocate(A);
+        continue;
```

Comment:
I am confused by this. I don't know the details of the memory allocator class, but the comment reads "Frees the allocation associated with \p handle.". Therefore it seems strange that we would free something that we haven't run yet.
Could you add a comment explaining what this is doing?

Reply:
Comment added, this deallocation is more of a planned deallocation. It's a way of making sure we only allocate the memory we need at runtime.

Reply:
Hmm, that sounds weird.
I would have expected that we just deallocate all the activations and inputs/outputs at the end of running a network. Why do we need to register them?
Again not something to fix here. I see you're just moving things around.